src/rgw/rgw_lc.cc

```diff
@@ -1268,25 +1297,73 @@ class LCOpAction_Transition : public LCOpAction {
     /* If bucket is versioned, create delete_marker for current version
      */
     if (oc.bucket->versioned() && oc.o.is_current() && !oc.o.is_delete_marker()) {
-      ret = remove_expired_obj(oc.dpp, oc, false, rgw::notify::ObjectExpiration);
+      ret = remove_expired_obj(oc.dpp, oc, false, rgw::notify::ObjectTransitionCurrent);
       ldpp_dout(oc.dpp, 20) << "delete_tier_obj Object(key:" << oc.o.key << ") current & not delete_marker" << " versioned_epoch:  " << oc.o.versioned_epoch << "flags: " << oc.o.flags << dendl;
     } else {
```

Comment:
`else` case  here refers to both non-current objects and non-versioned objects. So it may be best to add another case here. 

if (!oc.bucket.versioned())  {
   ret = remove_expired_obj(oc.dpp, oc, true, rgw::notify::ObjectTransition);
} else if (oc.o.is_current() && !oc.o.is_delete_marker()) {
   ret = remove_expired_obj(oc.dpp, oc, false, rgw::notify::ObjectTransitionCurrent);
} else {
   ret = remove_expired_obj(oc.dpp, oc, true, rgw::notify::ObjectTransitionNoncurrent);
}

Reply:
thanks, let me revisit that

Reply:
I think the notification should still be ObjectTransitionCurrent?  that's how expiration works...